kornia/augmentation/_2d/intensity/random_rain.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.augmentation.random_generator._2d import RainGenerator
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class RandomRain(IntensityAugmentationBase2D):
+    r"""Add Random Rain to the image.
+
+    Args:
+        p: probability of applying the transformation.
+        number_of_drops: number of drops per image
+        drop_height: height of the drop in image(same for each drops in one image)
+        drop_width: width of the drop in image(same for each drops in one image)
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`, Optional: :math:`(B, 3, 3)`
+        - Output: :math:`(B, C, H, W)`
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.rand(1, 1, 5, 5)
+        >>> rain = RandomRain(p=1,drop_height=(1,2),drop_width=(1,2),number_of_drops=(1,1))
+        >>> rain(input)
+        tensor([[[[0.4963, 0.7682, 0.0885, 0.1320, 0.3074],
+                  [0.7843, 0.4901, 0.8964, 0.4556, 0.6323],
+                  [0.3489, 0.4017, 0.0223, 0.1689, 0.2939],
+                  [0.5185, 0.6977, 0.8000, 0.1610, 0.2823],
+                  [0.6816, 0.9152, 0.3971, 0.8742, 0.4194]]]])
+    """
+
+    def __init__(
+        self,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+        number_of_drops: Tuple[int, int] = (1000, 2000),
+        drop_height: Tuple[int, int] = (5, 20),
+        drop_width: Tuple[int, int] = (-5, 5),
+    ):
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+        self._param_generator = RainGenerator(number_of_drops, drop_height, drop_width)
+
+    def apply_transform(
+        self, image: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK(image.shape[1] == 3 or image.shape[1] == 1, "Number of color channels should be 1 or 3.")
+        KORNIA_CHECK(len(image.shape) in (3, 4), "Wrong input dimension.")
+
+        for i in range(image.shape[0]):
```

Comment:
Try avoiding for-loops here.

Reply:
Not sure this is possible, as we need to have a different number of drops on each image, and drops on each image should have different widths and heights.

Reply:
At least you may compute the height, width in a batch before applying.

Try avoiding for-loops and use vectorization as much as possible. I strongly concern about the speed of the current form.

Reply:
I try to compute all coordinates out of for loop, but due to different number_of_drops on each image, the algorithm works slower after vectorization.

Reply:
I did not get the reason. You mean that the algorithm is slower because the vectorization computes all drop "candidates", but it actually computes less if within the for-loop?